pkg/destroy/aws/aws.go

```diff
@@ -504,6 +525,65 @@ func deleteEC2Image(client *ec2.EC2, id string, logger logrus.FieldLogger) error
 		return err
 	}
 
+	logger.Info("Deleted")
+
+	// need to delete the snapshots after deleting the ami. aws will block the
+	// snapshot deletion with an InvalidSnapshot.InUse error until the ami is deleted.
+	return deleteEC2Snapshots(client, snapshots, logger)
+}
+
+func deleteEC2Snapshots(client *ec2.EC2, snapshots []string, logger logrus.FieldLogger) error {
+	failedSnapshotsCh := make(chan string, len(snapshots))
+
+	var wg sync.WaitGroup
+	wg.Add(len(snapshots))
+	for i, id := range snapshots {
+		go func(ix int, snapshotId string) {
+			defer wg.Done()
+			// make at most 10 attempts over around 5 minutes
+			err := wait.ExponentialBackoff(
+				wait.Backoff{
+					Duration: 5 * time.Second,
+					Factor:   1.45,
+					Jitter:   0.1,
+					Steps:    10,
+				},
+				func() (bool, error) {
+					err := deleteEC2Snapshot(client, snapshotId, logger.WithField("Snapshot", snapshotId))
+					return err == nil, nil
+				},
+			)
+			if err != nil {
+				failedSnapshotsCh <- snapshotId
+			}
+		}(i, id)
+	}
+	wg.Wait()
+
+	close(failedSnapshotsCh)
+	failedSnapshots := []string{}
+	for id := range failedSnapshotsCh {
+		failedSnapshots = append(failedSnapshots, id)
+	}
+	if len(failedSnapshots) > 0 {
+		return errors.Errorf("failed to delete snapshots %s", strings.Join(failedSnapshots, ", "))
+	}
+
+	return nil
+}
+
+func deleteEC2Snapshot(client *ec2.EC2, id string, logger logrus.FieldLogger) error {
```

Comment:
`deleteEC2Snapshot` alphabetizes between `deleteEC2SecurityGroup` and `deleteEC2Subnet`, so this function definition should be shifted down there.

Reply:
Done in c07b1ed to fcade11.